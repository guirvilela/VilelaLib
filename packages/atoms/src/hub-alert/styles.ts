import styled from 'styled-components';

import { HubAlertProps } from './types';

export const AlertInnerStyle = styled.div`
  display: flex;
  gap: ${({ theme }) => theme.spaces[2]};
  margin-top: -2px;
  width: 100%;
`;

export const AlertTitleStyle = styled.span`
  font-family: ${({ theme }) => theme.typography.body2.family};
  font-size: ${({ theme }) => theme.typography.body2.size};
  line-height: ${({ theme }) => theme.typography.body2.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body2.letterSpacing};
  color: ${({ theme }) => theme.colors.neutral[800]};
`;

export const AlertContentStyle = styled.span`
  font-family: ${({ theme }) => theme.typography.body2.family};
  font-size: ${({ theme }) => theme.typography.body2.size};
  line-height: ${({ theme }) => theme.typography.body2.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body2.letterSpacing};
  color: ${({ theme }) => theme.colors.neutralGrey[700]};
  font-weight: 400;
  width: 100%;
  text-overflow: ellipsis;
  overflow: hidden;
`;

export const AlertCloseStyle = styled.button`
  background-color: transparent;
  width: ${({ theme }) => theme.spaces[3]};
  height: ${({ theme }) => theme.spaces[3]};
  border: none;
  color: inherit;
  cursor: pointer;
`;

export const AlertStyle = styled.div<Omit<HubAlertProps, 'title'>>`
  display: flex;
  width: 100%;
  background-color: ${({ theme, color }) =>
    color ? theme.colors[color][100] : theme.colors.neutral[100]};
  color: ${({ theme, color }) =>
    color ? theme.colors[color][500] : theme.colors.neutral[500]};
  border: 1px solid
    ${({ theme, color }) =>
      color ? theme.colors[color][500] : theme.colors.neutral[500]};
  border-radius: ${({ theme, radius }) =>
    radius ? theme.spaces[radius] : '4px'};
  align-items: ${({ inline }) => (inline ? 'center' : 'start')};

  ${({ theme, pad }) => {
    if (typeof pad === 'string') {
      return `padding: ${theme.spaces[pad]};`;
    }
    if (Array.isArray(pad)) {
      return `padding: ${pad.reduce(
        (acc, x) => acc + `${theme.spaces[x]} `,
        ''
      )};`;
    }
    return `padding: ${theme.spaces[2]};`;
  }}

  gap: ${({ theme, gap }) => (gap ? theme.spaces[gap] : theme.spaces[2])};

  ${AlertInnerStyle} {
    flex-direction: ${({ inline }) => (inline ? 'row' : 'column')};
    align-items: ${({ inline }) => (inline ? 'center' : 'start')};
  }

  ${AlertTitleStyle} {
    font-weight: ${({ inline }) => (inline ? 400 : 600)};
    white-space: ${({ nowrap }) => (nowrap ? 'nowrap' : undefined)};
  }
`;
