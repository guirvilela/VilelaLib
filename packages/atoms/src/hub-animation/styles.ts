import styled from 'styled-components';

import { HubAnimationProps } from './types';

export const AnimationStyle = styled.div<HubAnimationProps & { ref }>`
  ${({ theme, size, width, height }) => {
    switch (size) {
      case 'xs':
        return `
          width: ${theme.spaces[2]};
          height: ${theme.spaces[2]};
          min-width: ${theme.spaces[2]};
        `;
      case 'sm':
        return `
          width: ${theme.spaces[3]};
          height: ${theme.spaces[3]};
          min-width: ${theme.spaces[3]};
        `;
      case 'md':
        return `
          width: ${theme.spaces[5]};
          height: ${theme.spaces[5]};
          min-width: ${theme.spaces[5]};
        `;
      case 'lg':
        return `
          width: ${theme.spaces[6]};
          height: ${theme.spaces[6]};
          min-width: ${theme.spaces[6]};
        `;
      default:
        return `
          width: ${width || theme.spaces[5]};
          height: ${height || theme.spaces[5]};
          min-width: ${width || theme.spaces[5]};
        `;
    }
  }}
`;
