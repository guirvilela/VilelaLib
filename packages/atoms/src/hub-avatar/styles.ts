import styled from 'styled-components';

import { HubAvatarProps } from './types';

export const AvatarImgStyle = styled.img`
  width: 100%;
  height: 100%;
  background-size: cover;
  border-radius: 50%;
`;

export const AvatarInitialsStyle = styled.span`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  border-radius: 50%;
`;

export const AvatarStyle = styled.div<HubAvatarProps>`
  ${({ theme, size }) => {
    if (!size) {
      return `
        width: ${theme.spaces[7]};
        height: ${theme.spaces[7]};
        min-width: ${theme.spaces[7]};
      `;
    }

    switch (size) {
      case 'xs':
        return `
          width: ${theme.spaces[3]};
          height: ${theme.spaces[3]};
          min-width: ${theme.spaces[3]};
        `;
      case 'sm':
        return `
          width: ${theme.spaces[5]};
          height: ${theme.spaces[5]};
          min-width: ${theme.spaces[5]};
        `;
      case 'md':
        return `
          width: ${theme.spaces[7]};
          height: ${theme.spaces[7]};
          min-width: ${theme.spaces[7]};
        `;
      case 'lg':
        return `
          width: ${theme.spaces[10]};
          height: ${theme.spaces[10]};
          min-width: ${theme.spaces[10]};
        `;
      default:
        return `
          width: ${size};
          height: ${size};
          min-width: ${size};
        `;
    }
  }}
  ${({ borderless, theme }) =>
    borderless
      ? ''
      : `
    padding: ${theme.spaces[1]};
    border: 1px solid ${theme.colors.neutral[400]};
  `};

  border-radius: 50%;

  ${AvatarInitialsStyle} {
    ${({ theme, typo }) => `
      font-family: ${theme.typography[typo || 'h4'].family};
      font-size: ${theme.typography[typo || 'h4'].size};
      line-height: ${theme.typography[typo || 'h4'].lineHeight};
      letter-spacing: ${theme.typography[typo || 'h4'].letterSpacing};
    `}

    background-color: ${({ theme, color, colorLevel }) =>
      color && colorLevel
        ? theme.colors[color][colorLevel]
        : color && theme.colors[color]
        ? theme.colors[color][500]
        : color
        ? color
        : undefined};

    color: ${({ theme, colorText, colorTextLevel }) =>
      colorText && colorTextLevel
        ? theme.colors[colorText][colorTextLevel]
        : colorText && theme.colors[colorText]
        ? theme.colors[colorText][500]
        : colorText
        ? colorText
        : theme.colors.neutral[800]};
  }
`;
