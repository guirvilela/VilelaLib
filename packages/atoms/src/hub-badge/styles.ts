import styled, { css } from 'styled-components';

import { HubBadgeProps } from './types';

const allStyle = css<HubBadgeProps>`
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  gap: ${({ theme }) => theme.spaces[2]};

  ${({ theme, size, width, height }) => {
    if (width || height) {
      return `
        width: ${width};
        height: ${height};
      `;
    }

    switch (size) {
      case 'xs':
        return `
          width: ${theme.spaces[2]};
          height: ${theme.spaces[2]};
        `;
      case 'sm':
        return `
          width: ${theme.spaces[3]};
          height: ${theme.spaces[3]};
        `;
      case 'md':
        return `
          width: ${theme.spaces[4]};
          height: ${theme.spaces[4]};
        `;
      case 'lg':
        return `
          width: ${theme.spaces[5]};
          height: ${theme.spaces[5]};
        `;
      default:
        return '';
    }
  }}

  background-color: ${({ theme, color, colorLevel }) =>
    color && colorLevel
      ? theme.colors[color][colorLevel]
      : color
      ? theme.colors[color][100]
      : theme.colors.neutral[200]};

  color: ${({ theme, color, textColor, textColorLevel }) =>
    textColor && textColorLevel
      ? theme.colors[textColor][textColorLevel]
      : textColor
      ? theme.colors[textColor][500]
      : color
      ? theme.colors[color][500]
      : theme.colors.neutral[500]};

  border-radius: ${({ theme, rounded, radius }) =>
    rounded ? '50%' : theme.spaces[radius || '1']};

  ${({ theme, pad }) => {
    if (!pad) return `padding: 2px 6px`;
    if (typeof pad === 'string') return `padding: ${theme.spaces[pad]}`;
    if (pad.length === 4) {
      return `padding: ${theme.spaces[pad[0]]} ${theme.spaces[pad[1]]} ${
        theme.spaces[pad[2]]
      } ${theme.spaces[pad[3]]}`;
    }
    if (pad.length === 3) {
      return `padding: ${theme.spaces[pad[0]]} ${theme.spaces[pad[1]]} ${
        theme.spaces[pad[2]]
      }`;
    }
    if (pad.length === 2) {
      return `padding: ${theme.spaces[pad[0]]} ${theme.spaces[pad[1]]}`;
    }
    return `padding: ${theme.spaces[pad[0]]}`;
  }};

  font-family: ${({ theme }) => theme.typography.body2.family};
  font-size: ${({ theme }) => theme.typography.body2.size};
  line-height: ${({ theme }) => theme.typography.body2.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body2.letterSpacing};
  font-weight: 600;
`;

const floatStyle = css<HubBadgeProps>`
  position: relative;

  &::after {
    ${allStyle}

    position: absolute;
    content: '${({ content }) => content}';
    border: 1px solid ${({ theme }) => theme.colors.white[500]};
    ${({ theme, position }) => {
      if (position) {
        if (Array.isArray(position)) {
          return `
            top: -${theme.spaces[position[0]]};
            right: -${theme.spaces[position[1]]};
          `;
        }
        return `
          top: -${theme.spaces[position]};
          right: -${theme.spaces[position]};
        `;
      }
      return `
        top: -10px;
        right: -10px;
      `;
    }}
    ${({ theme }) => `
      min-width: ${theme.spaces[4]};
      width: ${theme.spaces[4]};
      height: ${theme.spaces[4]};
    `}
    padding: 0;

    font-family: ${({ theme }) => theme.typography.body3.family};
    font-size: ${({ theme }) => theme.typography.body3.size};
    line-height: ${({ theme }) => theme.typography.body3.lineHeight};
    letter-spacing: ${({ theme }) => theme.typography.body3.letterSpacing};
  }
`;

export const BadgeStyle = styled.div<HubBadgeProps>`
  ${({ float, content }) => {
    if (float) {
      if (content) {
        return floatStyle;
      }
    } else {
      return allStyle;
    }
    return '';
  }}
`;
