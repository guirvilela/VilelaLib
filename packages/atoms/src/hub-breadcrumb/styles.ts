import styled from 'styled-components';

import { HubBreadcrumbProps, HubBreadcrumbsProps } from './types';

export const BreadcrumbsStyle = styled.div<HubBreadcrumbsProps>`
  display: flex;
  align-items: center;
  overflow: auto;
`;

export const BreadcrumbsPathStyle = styled.div`
  display: flex;
  align-items: center;
`;

export const BreadcrumbsIconStyle = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  color: ${({ theme }) => theme.colors.neutral[400]};
  margin: ${({ theme }) => `0 ${theme.spaces[2]}`};
`;

export const BreadcrumbStyle = styled.span<HubBreadcrumbProps>`
  display: flex;
  align-items: center;
  gap: ${({ theme }) => theme.spaces[2]};
  font-family: ${({ theme }) => theme.typography.body2.family};
  font-size: ${({ theme }) => theme.typography.body2.size};
  line-height: ${({ theme }) => theme.typography.body2.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body2.letterSpacing};
  color: ${({ theme, active }) =>
    active ? theme.colors.primary[500] : theme.colors.neutral[400]};
  transition: color 0.5s;
  font-weight: 600;
  white-space: ${({ nowrap }) => (nowrap ? 'nowrap' : 'pre-wrap')};

  ${({ width }) =>
    width
      ? `
    display: block;
    max-width: ${width};
    text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap;
  `
      : ''}

  &:hover {
    color: ${({ theme }) => theme.colors.primary[500]};
  }
`;
