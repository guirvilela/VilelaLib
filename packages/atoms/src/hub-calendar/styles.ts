import styled from 'styled-components';

import { HubCalendarItem, HubCalendarProps } from './types';

export const CalendarHeadStyle = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: ${({ theme }) => theme.spaces[3]};
  width: 100%;
`;

export const CalendarHeadTitleStyle = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  gap: ${({ theme }) => theme.spaces[2]};

  font-family: ${({ theme }) => theme.typography.h4.family};
  font-size: ${({ theme }) => theme.typography.h4.size};
  line-height: ${({ theme }) => theme.typography.h4.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.h4.letterSpacing};
  font-weight: 500;
`;

export const CalendarHeadLinkStyle = styled.button`
  background-color: transparent;
  color: ${({ theme }) => theme.colors.primary[200]};
  border: none;

  :hover {
    text-decoration: underline;
    cursor: pointer;
  }
`;

export const CalendarHeadNextStyle = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  background-color: ${({ theme }) => theme.colors.neutral[200]};
  border-radius: 50%;
  cursor: pointer;

  height: ${({ theme }) => theme.spaces[6]};
  width: ${({ theme }) => theme.spaces[6]};
`;

export const CalendarHeadPrevStyle = styled(CalendarHeadNextStyle)`
  transform: rotate(-180deg);
`;

export const CalendarDaysStyle = styled.div`
  display: grid;
  align-items: center;
  justify-content: center;
  justify-items: center;
  gap: ${({ theme }) => theme.spaces[2]};
  grid-template-columns: auto auto auto auto auto auto auto;
  width: 100%;
`;

export const CalendarMonthsStyle = styled.div`
  display: grid;
  grid-template-columns: 50% 50%;
  width: 100%;
`;

export const CalendarYearsStyle = styled.div`
  display: grid;
  grid-template-columns: 50% 50%;
  width: 100%;
`;

export const CalendarPeriodStyle = styled.span`
  display: flex;
  align-items: center;
  justify-content: center;

  font-family: ${({ theme }) => theme.typography.body3.family};
  font-size: ${({ theme }) => theme.typography.body3.size};
  line-height: ${({ theme }) => theme.typography.body3.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body3.letterSpacing};
  font-weight: 400;
`;

export const CalendarDayStyle = styled.button<HubCalendarItem>`
  display: flex;
  align-items: center;
  justify-content: center;

  color: ${({
    theme,
    selected,
    active,
    previous,
    next,
    disabled,
    today,
    event
  }) => {
    if (selected || active) {
      if (previous || next) {
        if (event && event.color) {
          return theme.colors[event.color][300];
        }
        return theme.colors.neutral[500];
      }
      return theme.colors.white[500];
    }
    if (previous || next) {
      if (event && event.color) {
        return theme.colors[event.color][200];
      }
      return theme.colors.neutral[300];
    }
    if (disabled) {
      if (event && event.color) {
        return theme.colors[event.color][400];
      }
      return theme.colors.neutral[300];
    }
    if (today || (event && event.color)) {
      if (event && event.color) {
        return theme.colors[event.color][500];
      }
      return theme.colors.info[600];
    }
    return theme.colors.neutral[800];
  }};
  background-color: ${({
    theme,
    selected,
    active,
    today,
    event,
    previous,
    next
  }) => {
    if (previous || next) {
      if (event && event.color) {
        return theme.colors[event.color][400];
      }
    }
    if (selected) {
      if (event && event.color) {
        return theme.colors[event.color][500];
      }
      return theme.colors.info[500];
    }
    if (active) {
      if (event && event.color) {
        return theme.colors[event.color][300];
      }
      return theme.colors.info[300];
    }
    if (today || (event && event.color)) {
      if (event && event.color) {
        return theme.colors[event.color][100];
      }
      return theme.colors.info[100];
    }
    return 'transparent';
  }};
  border-radius: 50%;
  border: none;
  ${({ disabled }) => (disabled ? 'cursor: not-allowed;' : `cursor: pointer;`)};
  transition: 0.5s ease background-color, 0.2s ease color;
`;

export const CalendarMonthYearStyle = styled.button<HubCalendarItem>`
  display: flex;
  align-items: center;
  justify-content: center;
  padding: ${({ theme }) => `${theme.spaces[1]} ${theme.spaces[2]}`};
  background-color: transparent;
  color: ${({ theme, selected, active }) => {
    if (selected) {
      return theme.colors.info[500];
    }
    if (!active) {
      return theme.colors.neutral[300];
    }
    return theme.colors.neutral[800];
  }};
  border-radius: 50%;
  border: none;
  cursor: pointer;

  :hover {
    text-decoration: underline;
  }
`;

export const CalendarActionsStyle = styled.div`
  display: grid;
  align-items: center;
  justify-content: end;
  gap: ${({ theme }) => theme.spaces[2]};
  width: 100%;
  padding: ${({ theme }) => `0 ${theme.spaces[2]}`};
`;

export const CalendarTodayStyle = styled.button`
  border: none;
  background-color: transparent;
  color: ${({ theme, disabled }) => {
    if (disabled) {
      return theme.colors.neutral[300];
    }
    return theme.colors.primary[200];
  }};
  ${({ disabled }) => (disabled ? '' : `cursor: pointer;`)};
`;

export const CalendarStyle = styled.div<HubCalendarProps>`
  display: flex;
  align-items: center;
  flex-direction: column;
  gap: ${({ theme }) => theme.spaces[2]};
  width: ${({ width }) => width || '100%'};

  ${CalendarDayStyle} {
    ${({ theme, size }) => {
      switch (size) {
        case 'xs':
          return `width: ${theme.spaces[2]}; height: ${theme.spaces[2]};`;
        case 'sm':
          return `width: ${theme.spaces[3]}; height: ${theme.spaces[3]};`;
        case 'md':
          return `width: ${theme.spaces[5]}; height: ${theme.spaces[5]};`;
        case 'lg':
          return `width: ${theme.spaces[6]}; height: ${theme.spaces[6]};`;
        default:
          if (size) {
            return `width: ${size}; height: ${size};`;
          }
          return `width: ${theme.spaces[6]}; height: ${theme.spaces[6]};`;
      }
    }};
  }

  ${CalendarTodayStyle},
  ${CalendarDayStyle} {
    font-family: ${({ theme, typo }) => theme.typography[typo || 'h4'].family};
    font-size: ${({ theme, typo }) => theme.typography[typo || 'h4'].size};
    line-height: ${({ theme, typo }) =>
      theme.typography[typo || 'h4'].lineHeight};
    letter-spacing: ${({ theme, typo }) =>
      theme.typography[typo || 'h4'].letterSpacing};
    font-weight: ${({ typo }) =>
      typo ? (typo.includes('body') ? 600 : 400) : 400};
  }

  ${CalendarHeadTitleStyle},
  ${CalendarHeadLinkStyle} {
    font-family: ${({ theme, typo, typoHeader }) =>
      theme.typography[typoHeader || typo || 'h4'].family};
    font-size: ${({ theme, typo, typoHeader }) =>
      theme.typography[typoHeader || typo || 'h4'].size};
    line-height: ${({ theme, typo, typoHeader }) =>
      theme.typography[typoHeader || typo || 'h4'].lineHeight};
    letter-spacing: ${({ theme, typo, typoHeader }) =>
      theme.typography[typoHeader || typo || 'h4'].letterSpacing};
    font-weight: ${({ typo, typoHeader }) =>
      typoHeader || typo
        ? (typoHeader || typo)?.includes('body')
          ? 600
          : 400
        : 400};
  }
`;
