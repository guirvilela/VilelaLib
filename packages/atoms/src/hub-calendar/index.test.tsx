import { I18nProvider } from '@hubert/i18n';
import { wrapse } from '@hubert/shared';
import { ThemeProvider } from '@hubert/styles';
import { render } from '@testing-library/react';
import React from 'react';

import { HubCalendar } from '.';

describe('HubCalendar component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<HubCalendar />, {
      wrapper: wrapse(ThemeProvider, I18nProvider)
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
