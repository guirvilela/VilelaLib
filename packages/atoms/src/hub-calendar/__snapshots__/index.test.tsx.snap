// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HubCalendar component matches snapshot 1`] = `
<DocumentFragment>
  .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  gap: 16px;
  width: 100%;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 8px;
  font-family: Kanit;
  font-size: 20px;
  line-height: 1.4;
  -webkit-letter-spacing: 0.01em;
  -moz-letter-spacing: 0.01em;
  -ms-letter-spacing: 0.01em;
  letter-spacing: 0.01em;
  font-weight: 500;
}

.c7 {
  background-color: transparent;
  color: #98AAC1;
  border: none;
}

.c7:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  background-color: #E8EEF4;
  border-radius: 50%;
  cursor: pointer;
  height: 32px;
  width: 32px;
}

.c3 {
  -webkit-transform: rotate(-180deg);
  -ms-transform: rotate(-180deg);
  transform: rotate(-180deg);
}

.c8 {
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-items: center;
  gap: 8px;
  grid-template-columns: auto auto auto auto auto auto auto;
  width: 100%;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Nunito Sans;
  font-size: 12px;
  line-height: 1.6666666666666667;
  -webkit-letter-spacing: 0.01em;
  -moz-letter-spacing: 0.01em;
  -ms-letter-spacing: 0.01em;
  letter-spacing: 0.01em;
  font-weight: 400;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #CCD5DD;
  background-color: transparent;
  border-radius: 50%;
  border: none;
  cursor: pointer;
  -webkit-transition: 0.5s ease background-color,0.2s ease color;
  transition: 0.5s ease background-color,0.2s ease color;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #1E71AE;
  background-color: #EEF8FF;
  border-radius: 50%;
  border: none;
  cursor: pointer;
  -webkit-transition: 0.5s ease background-color,0.2s ease color;
  transition: 0.5s ease background-color,0.2s ease color;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #040C12;
  background-color: transparent;
  border-radius: 50%;
  border: none;
  cursor: pointer;
  -webkit-transition: 0.5s ease background-color,0.2s ease color;
  transition: 0.5s ease background-color,0.2s ease color;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 8px;
  width: 100%;
}

.c0 .c10 {
  width: 32px;
  height: 32px;
}

.c0 .c4,
.c0 .c6 {
  font-family: Kanit;
  font-size: 20px;
  line-height: 1.4;
  -webkit-letter-spacing: 0.01em;
  -moz-letter-spacing: 0.01em;
  -ms-letter-spacing: 0.01em;
  letter-spacing: 0.01em;
  font-weight: 400;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <button
        class="c2 c3"
        type="button"
      >
        <svg
          fill="none"
          height="12"
          viewBox="0 0 7 12"
          width="7"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6.46677 5.48313L2.23243 1.2575C2.1396 1.16409 2.02914 1.08995 1.90745 1.03935C1.78575 0.988757 1.65522 0.962708 1.52338 0.962708C1.39155 0.962708 1.26102 0.988757 1.13932 1.03935C1.01762 1.08995 0.907169 1.16409 0.81433 1.2575C0.628328 1.44423 0.523926 1.69682 0.523926 1.96011C0.523926 2.2234 0.628328 2.476 0.81433 2.66272L4.3496 6.19072L0.81433 9.71872C0.628328 9.90545 0.523926 10.158 0.523926 10.4213C0.523926 10.6846 0.628328 10.9372 0.81433 11.1239C0.907645 11.2163 1.01831 11.2894 1.13999 11.339C1.26166 11.3886 1.39195 11.4137 1.52338 11.413C1.65481 11.4137 1.7851 11.3886 1.90678 11.339C2.02845 11.2894 2.13912 11.2163 2.23243 11.1239L6.46677 6.89831C6.56038 6.80567 6.63467 6.69544 6.68537 6.57399C6.73607 6.45255 6.76218 6.32229 6.76218 6.19072C6.76218 6.05916 6.73607 5.92889 6.68537 5.80745C6.63467 5.686 6.56038 5.57578 6.46677 5.48313Z"
            fill="#040C12"
          />
        </svg>
      </button>
      <div
        class="c4 c5"
      >
        <button
          class="c6 c7"
          type="button"
        >
          Janeiro
        </button>
        <button
          class="c6 c7"
          type="button"
        >
          1970
        </button>
      </div>
      <button
        class="c2"
        type="button"
      >
        <svg
          fill="none"
          height="12"
          viewBox="0 0 7 12"
          width="7"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6.46677 5.48313L2.23243 1.2575C2.1396 1.16409 2.02914 1.08995 1.90745 1.03935C1.78575 0.988757 1.65522 0.962708 1.52338 0.962708C1.39155 0.962708 1.26102 0.988757 1.13932 1.03935C1.01762 1.08995 0.907169 1.16409 0.81433 1.2575C0.628328 1.44423 0.523926 1.69682 0.523926 1.96011C0.523926 2.2234 0.628328 2.476 0.81433 2.66272L4.3496 6.19072L0.81433 9.71872C0.628328 9.90545 0.523926 10.158 0.523926 10.4213C0.523926 10.6846 0.628328 10.9372 0.81433 11.1239C0.907645 11.2163 1.01831 11.2894 1.13999 11.339C1.26166 11.3886 1.39195 11.4137 1.52338 11.413C1.65481 11.4137 1.7851 11.3886 1.90678 11.339C2.02845 11.2894 2.13912 11.2163 2.23243 11.1239L6.46677 6.89831C6.56038 6.80567 6.63467 6.69544 6.68537 6.57399C6.73607 6.45255 6.76218 6.32229 6.76218 6.19072C6.76218 6.05916 6.73607 5.92889 6.68537 5.80745C6.63467 5.686 6.56038 5.57578 6.46677 5.48313Z"
            fill="#040C12"
          />
        </svg>
      </button>
    </div>
    <div
      class="c8"
    >
      <span
        class="c9"
      >
        DOM.
      </span>
      <span
        class="c9"
      >
        SEG.
      </span>
      <span
        class="c9"
      >
        TER.
      </span>
      <span
        class="c9"
      >
        QUA.
      </span>
      <span
        class="c9"
      >
        QUI.
      </span>
      <span
        class="c9"
      >
        SEX.
      </span>
      <span
        class="c9"
      >
        S√ÅB.
      </span>
      <button
        class="c10 c11"
        label="28"
        type="button"
      >
        28
      </button>
      <button
        class="c10 c11"
        label="29"
        type="button"
      >
        29
      </button>
      <button
        class="c10 c11"
        label="30"
        type="button"
      >
        30
      </button>
      <button
        class="c10 c11"
        label="31"
        type="button"
      >
        31
      </button>
      <button
        class="c10 c12"
        label="1"
        title="Hoje"
        type="button"
      >
        1
      </button>
      <button
        class="c10 c13"
        label="2"
        type="button"
      >
        2
      </button>
      <button
        class="c10 c13"
        label="3"
        type="button"
      >
        3
      </button>
      <button
        class="c10 c13"
        label="4"
        type="button"
      >
        4
      </button>
      <button
        class="c10 c13"
        label="5"
        type="button"
      >
        5
      </button>
      <button
        class="c10 c13"
        label="6"
        type="button"
      >
        6
      </button>
      <button
        class="c10 c13"
        label="7"
        type="button"
      >
        7
      </button>
      <button
        class="c10 c13"
        label="8"
        type="button"
      >
        8
      </button>
      <button
        class="c10 c13"
        label="9"
        type="button"
      >
        9
      </button>
      <button
        class="c10 c13"
        label="10"
        type="button"
      >
        10
      </button>
      <button
        class="c10 c13"
        label="11"
        type="button"
      >
        11
      </button>
      <button
        class="c10 c13"
        label="12"
        type="button"
      >
        12
      </button>
      <button
        class="c10 c13"
        label="13"
        type="button"
      >
        13
      </button>
      <button
        class="c10 c13"
        label="14"
        type="button"
      >
        14
      </button>
      <button
        class="c10 c13"
        label="15"
        type="button"
      >
        15
      </button>
      <button
        class="c10 c13"
        label="16"
        type="button"
      >
        16
      </button>
      <button
        class="c10 c13"
        label="17"
        type="button"
      >
        17
      </button>
      <button
        class="c10 c13"
        label="18"
        type="button"
      >
        18
      </button>
      <button
        class="c10 c13"
        label="19"
        type="button"
      >
        19
      </button>
      <button
        class="c10 c13"
        label="20"
        type="button"
      >
        20
      </button>
      <button
        class="c10 c13"
        label="21"
        type="button"
      >
        21
      </button>
      <button
        class="c10 c13"
        label="22"
        type="button"
      >
        22
      </button>
      <button
        class="c10 c13"
        label="23"
        type="button"
      >
        23
      </button>
      <button
        class="c10 c13"
        label="24"
        type="button"
      >
        24
      </button>
      <button
        class="c10 c13"
        label="25"
        type="button"
      >
        25
      </button>
      <button
        class="c10 c13"
        label="26"
        type="button"
      >
        26
      </button>
      <button
        class="c10 c13"
        label="27"
        type="button"
      >
        27
      </button>
      <button
        class="c10 c13"
        label="28"
        type="button"
      >
        28
      </button>
      <button
        class="c10 c13"
        label="29"
        type="button"
      >
        29
      </button>
      <button
        class="c10 c13"
        label="30"
        type="button"
      >
        30
      </button>
      <button
        class="c10 c13"
        label="31"
        type="button"
      >
        31
      </button>
    </div>
  </div>
</DocumentFragment>
`;
